docs/antrea-network-policy.md

```diff
@@ -1106,6 +1113,84 @@ This clusterIP based match has one caveat: directly access to the Endpoints of t
 
+## ServiceAccount based selection
+
+Antrea ClusterNetworkPolicy accepts a `serviceAccounts` field to select all Pods that have been assigned one of
+ServiceAccounts selected by this field. This field could be used in `appliedTo`, ingress `from` and egress `to` section.
+No matter `serviceAccounts` is used in which sections, it cannot be used with any other fields.
+
+`serviceAccounts` can be used in three ways:
+
+1. Use `namespacedName`: The ServiceAccount with a specific name under a specific namespace will be selected.
+2. Use `objectSelector` in `namespacedObjectSelector`: All ServiceAccounts with specific labels in all namespaces will
+   be selected.
+3. Use `objectSelector`+`namespaceSelector` in `namespacedObjectSelector`: All ServiceAccounts with specific labels in
+   all namespaces that have
+   specific labels will be selected.
+
+An example policy using `serviceAccounts` could look like this:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: acnp-service-account
+spec:
+    priority: 5
+    tier: securityops
+    appliedTo:
+      - serviceAccounts:
+          namespacedObjectSelector:
+            objectSelector:
+              matchLabels:
+                level: user  
+    egress:
+      - action: Drop
+        to:
+          - serviceAccounts:
+              namespacedName:
+                name: sa-1
+                namespace: ns-1
+        name: ServiceAccountsEgressRule
+        enableLogging: false
+    ingress:
+     - action: Reject
+       from:
+         - serviceAccounts:
+             namespacedObjectSelector:
+               objectSelector:
+                 matchLabels:
+                   level: admin
+               namespaceSelector:
+                 matchLabels:
+                   env: prod
+       name: ServiceAccountsIngressRule
+       enableLogging: false
+```
+
+In this example, the policy will apply to all Pods whose ServiceAccount has label `level: user`.
+Let's call those Pods "appliedToPods".
+The egress `to` section will select all Pods whose ServiceAccount is in `ns-1` namespace and name as `sa-1`.
+Let's call those Pods "egressPods".
+The ingress `from` section will select all Pods whose ServiceAccount has label `level: admin` and at the same time,
+the ServiceAccount is in one of namespaces that has label `env: prod`.
+Let's call those Pods "ingressPods".
+
+So the policy will act as:
+
+Traffic from "appliedToPods" to "egressPods" will be dropped.
+Traffic from "ingressPods" to "appliedToPods" will be rejected.
+
+There are some CAVEATS after introducing `serviceAccounts`:
+
+Antrea ClusterNetworkPolicy requires `appliedTo` should be set either in `spec.appliedTo` or in `appliedTo` of all rules.
+If `serviceAccounts` field doesn't select any ServiceAccounts or selected ServiceAccounts are not used by any Pods,
+the policy will violate the requirement above. In this situation, the whole policy won't be processed.
```

Comment:
I thought a policy with a podSelector that selects nothing, an internalNP will still be created for it. When a Pod that matched this podSelector is created, this policy will start to be enforced on this Pod. But if `serviceAccount` doesn't select any ServiceAccount with label selector. We couldn't create pSel+nsSel for it. And if this cause `appliedTo` in `spec.appliedTo` and in `appliedTo` of all rules are nil, then it will be a totally no-op on this policy.

But anyway, after removing label selector in `serviceAccount` this won't be a problem. We can always translate `serviceAccount` to pSel+nsSel. After the translation, the rest part will be the same as the previous logic.